internal/tests/user_handler_test.go

```diff
@@ -0,0 +1,103 @@
+package tests
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"github.com/vigiloauth/vigilo/internal/users"
+	"github.com/vigiloauth/vigilo/pkg/identity/server"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+const (
+	username string = "username"
+	email    string = "email@email.com"
+	password string = "password"
+)
+
+func TestUserHandler_HandleUserRegistration(t *testing.T) {
+	tests := []struct {
+		name           string
+		requestBody    users.UserRegistrationRequest
+		expectedStatus int
+		wantError      bool
+	}{
+		{
+			name: "Successful User Registration",
+			requestBody: users.UserRegistrationRequest{
+				Username: username,
+				Email:    email,
+				Password: password,
+			},
+			expectedStatus: http.StatusCreated,
+			wantError:      false,
+		},
+		{
+			name: "User Registration fails given invalid request body",
+			requestBody: users.UserRegistrationRequest{
+				Username: "",
+				Email:    "invalidemail",
+				Password: "short",
+			},
+			expectedStatus: http.StatusBadRequest,
+			wantError:      true,
+		},
+		{
+			name: "Missing required fields in request",
+			requestBody: users.UserRegistrationRequest{
+				Username: "missingemailuser",
+			},
+			expectedStatus: http.StatusBadRequest,
+			wantError:      true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			body, err := json.Marshal(test.requestBody)
+			if err != nil {
+				t.Fatalf("failed to  marshal request body: %v", err)
+			}
+
+			rr := setupIdentityServer(body)
+			if rr.Code != test.expectedStatus {
+				t.Errorf("expected status %v, got %v", test.expectedStatus, rr.Code)
+			}
+
+			if test.wantError {
+				var responseBody map[string]interface{}
```

Comment:
Let's pull this into a method with a nice name to help the test be easy to understand.